client/cypress/integration/dashboard/parameter_mapping_spec.js

```diff
@@ -0,0 +1,136 @@
+import { createDashboard } from '../../support/redash-api';
+import { createQueryAndAddWidget } from '../../support/dashboard';
+
+describe('Parameter Mapping', () => {
+  beforeEach(function () {
+    cy.login();
+    createDashboard('Foo Bar').then(({ slug, id }) => {
+      this.dashboardId = id;
+      this.dashboardUrl = `/dashboard/${slug}`;
+    }).then(() => {
+      const queryData = {
+        name: 'Text Parameter',
+        query: "SELECT '{{test-parameter}}' AS parameter",
+        options: {
+          parameters: [
+            { name: 'test-parameter', title: 'Test Parameter', type: 'text', value: 'example' },
+          ],
+        },
+      };
+      const widgetOptions = { position: { col: 0, row: 0, sizeX: 3, sizeY: 10, autoHeight: false } };
+      createQueryAndAddWidget(this.dashboardId, queryData, widgetOptions).then((widgetTestId) => {
+        cy.visit(this.dashboardUrl);
+        this.widgetTestId = widgetTestId;
+      });
+    });
+  });
+
+  const openMappingOptions = (widgetTestId, paramName) => {
+    cy.getByTestId(widgetTestId)
+      .within(() => {
+        cy.getByTestId('WidgetDropdownButton')
+          .click();
+      });
+
+    cy.getByTestId('WidgetDropdownButtonMenu')
+      .contains('Edit Parameters')
+      .click();
+
+    cy.getByTestId(`EditParamMappingButon-${paramName}`)
+      .click();
+  };
+
+  const saveMappingOptions = () => {
+    cy.getByTestId('EditParamMappingPopover')
+      .within(() => {
+        cy.contains('button', 'OK')
+          .click();
+      });
+
+    cy.contains('button', 'OK')
+      .click();
+  };
+
+  it('has widget mapping as the default', function () {
```

Comment:
The default mapping by API is widget level param (cause api-redash:addWidget doesn't send parameterMapping data), but in the UI (AddWidgetDialog) it's dashboard level param.

Dunno if this is an actual problem or not, but it did confuse me when I read the test.

Reply:
Well noticed, I've never realized that :joy:. I'll rename the test for now.